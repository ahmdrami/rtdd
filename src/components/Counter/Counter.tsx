import React, { useState } from 'react'

const Counter: React.FunctionComponent = () => {
  const [counter, setCounter] = useState(0)
  return (
    <>
      <button onClick={() => setCounter(counter - 1)}>Minus</button>
      Clicked {counter} times!
      <button onClick={() => setCounter(counter + 1)}>Add</button>
    </>
  )
}

export default Counter
