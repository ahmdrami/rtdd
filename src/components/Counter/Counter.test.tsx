import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import Counter from './Counter'

describe('Counter component', () => {
  test('Initial value to be 0', () => {
    render(<Counter />)
    expect(screen.getByText(/^Clicked 0 times!$/i)).toBeInTheDocument()
  })

  test('Counter value decrements on clicking Minus button', () => {
    render(<Counter />)
    fireEvent.click(screen.getByText(/Minus/))
    expect(screen.getByText(/^Clicked -1 times!$/i)).toBeInTheDocument()

  })

  test('Counter value increments on clicking Add button', () => {
    render(<Counter />)
    fireEvent.click(screen.getByText(/Add/))
    expect(screen.getByText(/^Clicked 1 times!$/i)).toBeInTheDocument()

  })
})
