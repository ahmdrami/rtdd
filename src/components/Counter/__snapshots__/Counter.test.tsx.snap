// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Snapshot 1`] = `
<DocumentFragment>
  <button>
    Minus
  </button>
  Clicked 0 times!
  <button>
    Add
  </button>
</DocumentFragment>
`;
